src/Models/AdminUser.php

```diff
@@ -54,7 +54,7 @@ class AdminUser extends Authenticatable implements CanActivateContract, HasMedia
         'deleted_at',
     ];
 
-    protected $appends = ['full_name', 'resource_url'];
+    protected $appends = ['full_name', 'resource_url', 'avatar_thumb_url'];
```

Comment:
Nice & clean solution.

But it has a drawback - everytime user retrieves a AdminUser (i.e. collection), we end up quering media (even if someone wants to list only full_name (i.e. in selectbox).

What do you think @dejwCake ?

Reply:
I agree with @palypster, the drawback is significant, this approach is good if with every request for user you also need thumb, however this is not the case. 

But solution for this could be caching of avatar. So if you stay with this solution cash the avatar for long period of time and invalidate cache only when user changes his avatar.